### ISSUE TITLE
postcss-bem-linter: end with version 8.4.0 not working anymore

### ISSUE THREAD

**Issue by thomasklein** [stylelint/stylelint#3070]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug in combination with https://github.com/simonsmith/stylelint-selector-bem-pattern @simonsmith 

> Which rule, if any, is this issue related to?

Together with plugin stylelint-selector-bem-pattern but only since version 8.4.0, former versions 8.3.x worked.

> What CSS is needed to reproduce this issue?

e.g.

```css
/** postcss-bem-linter: define mycomp */

...some selector

/* postcss-bem-linter: end */

/* errors with "Invalid component selector (plugin/selector-bem-pattern)" although it should not be evaluated anymore with the plugin. Same result with /** @end */ */
.mycomp + #invalid-bem-selector + h2
```

> What stylelint configuration is needed to reproduce this issue?

Any valid configuration of "plugin/selector-bem-pattern": {

> Which version of stylelint are you using?

e.g. `8.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config myconfig *.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
test.css
 143:5  ✖  Invalid component selector ".mycomp + #invalid-bem-selector + h2"   plugin/selector-bem-pattern
```

**Comment by hudochenkov** [comment]
Sorry, I don't see how plugin violations are related to stylelint. Could you explain more, please, what's happening and why it shouldn't?

**Comment by thomasklein** [comment]
Hi @hudochenkov 
Sure. I updated stylelint from 8.3.1 to 8.4.0 (no other stylelint related packages were updated) and I got this behaviour. Yes, it is an error in https://github.com/simonsmith/stylelint-selector-bem-pattern (I use 2.0.0) which is a wrapper around https://github.com/postcss/postcss-bem-linter (3.1.0 within dependencies of stylelint-selector-bem-pattern) and ultimately only a plugin of stylelint. However, since I only got this behaviour after the update (downgrading to stylelint@8.3.1 makes the problem disappear) I don't know in which repo to best place the error :)

**Comment by hudochenkov** [comment]
We might have some regression, because plugin started works differently, although it wasn't updated. Would you mind creating a repository with a minimal reproducible example, please?

**Comment by jeddy3** [comment]
@thomasklein I'm afraid I wasn't able to reproduce this locally. I use the `selector-bem-pattern` myself on my current project, and `/* postcss-bem-linter: end */` continued to work after updating to `stylelint@8.4`. As @hudochenkov request, can you create a repository with a minimal reproducible example please? Until then, I'll close this issue for now.

**Comment by thomasklein** [comment]
HI @jeddy3! Once I'm back at work I will give it another shoot. I suppose several things work together on my plate that ultimately produce this error. Nevertheless, thank you for pointing out that it works in your setup.
